tests/python/contrib/test_ethosu/test_replace_pooling.py

```diff
@@ -0,0 +1,158 @@
+    if ifm_layout == "NHWC":
+        ifm_stride_c = 1
+        ifm_stride_w = ifm_shape[3]
+        ifm_stride_h = ifm_shape[2] * ifm_shape[3]
+        ofm_height = (ifm_shape[1] - pool_shape[0] + padding[0] + padding[0]) // strides[0] + 1
+        ofm_width = (ifm_shape[2] - pool_shape[1] + padding[1] + padding[1]) // strides[1] + 1
+    else:
+        ifm_stride_w = 16
+        ifm_stride_c = 16 * ifm_shape[3]
+        ifm_stride_h = 16 * ifm_shape[2] * ifm_shape[3]
+        ofm_height = (ifm_shape[1] - pool_shape[0] + padding[0] + padding[0]) // strides[0] + 1
+        ofm_width = (ifm_shape[3] - pool_shape[1] + padding[1] + padding[1]) // strides[1] + 1
+
+    if ofm_layout == "NHWC":
+        ofm_stride_c = 1
+        ofm_stride_w = ofm_channels if ofm_width > 1 else 1
+        ofm_stride_h = ofm_channels * ofm_width if ofm_height > 1 else 1
+    else:
+        ofm_stride_w = 16
+        ofm_stride_c = 16 * ofm_width
+        ofm_stride_h = 16 * ofm_width * ((ofm_channels - 1) // 16 + 1)
+
+    serial_pooling = spec.SerialPooling(
+        ifm=spec.SerialFeatureMap(
+            data_type="int8",
+            height=ifm_shape[1],
+            width=ifm_shape[2] if ifm_layout == "NHWC" else ifm_shape[3],
+            channels=ofm_channels,
+            tile_height_0=ifm_shape[1],
+            tile_height_1=0,
+            tile_width_0=ifm_shape[2] if ifm_layout == "NHWC" else ifm_shape[3],
+            tile_address_0=0,
+            tile_address_1=0,
+            tile_address_2=0,
+            tile_address_3=0,
+            scale=1.0,
+            zero_point=0,
+            layout=ifm_layout,
+            stride_h=ifm_stride_h,
+            stride_w=ifm_stride_w,
+            stride_c=ifm_stride_c,
+        ),
+        ofm=spec.SerialFeatureMap(
+            data_type="int8",
+            height=ofm_height,
+            width=ofm_width,
+            channels=ofm_channels,
+            tile_height_0=ofm_height,
+            tile_height_1=0,
+            tile_width_0=ofm_width,
+            tile_address_0=0,
+            tile_address_1=0,
+            tile_address_2=0,
+            tile_address_3=0,
+            scale=1.0,
+            zero_point=0,
+            layout=ofm_layout,
+            stride_h=ofm_stride_h,
+            stride_w=ofm_stride_w,
+            stride_c=ofm_stride_c,
+        ),
+        pooling_type=pooling_type,
+        pool_shape=spec.SerialKernel(
+            width=pool_shape[1],
+            height=pool_shape[0],
+            stride_w=strides[1],
+            stride_h=strides[0],
+            dilation_w=1,
+            dilation_h=1,
+        ),
+        padding=spec.SerialPadding(
+            top=padding[0], left=padding[1], bottom=padding[2], right=padding[3]
+        ),
+        activation=spec.SerialActivation(
+            op=activation,
+            clip_min=10 if activation == "CLIP" else 0,
+            clip_max=100 if activation == "CLIP" else 0,
+        ),
```

Comment:
Can we add a separate test function for testing `CLIP` in isolation? I don't want this to land with untested code if possible.

Reply:
I have already addressed Elen's [comment](https://github.com/apache/tvm/pull/9384#discussion_r739093198) and now only the `CLIP` activation function is tested and no other activations.

Reply:
Shouldn't we at least test the other activations work in isolation?

Reply:
We don't support any other activation functions yet... We can't really test clip in isolation either since we don't support "solitary" clip, it is only supported through clamping functionality in the primitive operators, so we need an operator to accompany the clip. Or perhaps I misunderstood what you meant by "testing in isolation"...

Reply:
Aha yes I see what you mean and I agree, we should test the only case where activation is not CLIP, that is when it is NONE. It should be straightforward to do by having a parameter from activation that takes on either CLIP or NONE. The assert in the end should work for both cases. @NicolaLancellotti would you mind incorporating it into your next patch?

Reply:
I re-added  the `NONE` test